app/models/grda_warehouse/data_source.rb

```diff
@@ -568,6 +574,99 @@ def self.options_for_select(user:)
       end
   end
 
+  delegate :error_count_threshold_reached?, to: :import_threshold, allow_nil: true
+  ##
+  # Determines whether imports should be paused based on error thresholds.
+  #
+  # This method checks if an import threshold is set and whether the number of errors
+  # exceeds the allowed limit. If so, it returns `true`, indicating that imports should
+  # be paused.
+  #
+  # @param total [Integer] The total number of rows in the import file. Required.
+  # @param errors [Integer] The number of errors encountered. Required.
+  # @return [Boolean] `true` if imports should be paused due to errors, otherwise `false`.
+  #
+  memoize def pause_imports_with_errors?(total:, errors:)
+    return false unless import_threshold.present?
+    return false unless import_threshold.pause_on_error_threshold
+
+    error_count_threshold_reached?(total, errors)
+  end
+
+  def ever_pause_imports?
+    ever_pause_imports_with_errors? || ever_pause_imports_with_record_changes?
+  end
```

Comment:
this probably can be removed, it's called in HmisCsvImporter::Importer but doesn't really need to be